asset/asset.go

```diff
@@ -2077,35 +2111,74 @@ func (req *GroupKeyRequest) Validate() error {
 	return nil
 }
 
+// NewGroupPubKey derives a group key for the asset group based on the group key
+// request and the genesis asset ID.
+func (req *GroupKeyRequest) NewGroupPubKey(genesisAssetID ID) (btcec.PublicKey,
+	error) {
+
+	// If the external key is not specified, we will construct a version 0
```

Comment:
Or do we just want to center on a new default version?

Reply:
Here's how I see it:

The upside of keeping the default on V0 is that it's well tested and shifting to V1 wouldn't win anything unless the minter requires external signing.

The downside of keeping the default on V0 is that the debugging/user troubleshooting space is bigger.

The other downside is that not all our itests will cover V1 like they do V0.

I think, if we had the time, we should default to V1. And update all itests and unit tests to V1. The problem is that we still need almost the same test set for V0 because we will still support V0 going forwards.